core/templates/pages/topic-editor-page/modal-templates/create-new-subtopic-modal.component.spec.ts

```diff
@@ -0,0 +1,253 @@
+  let topicUpdateService: TopicUpdateService;
+  let topicEditorStateService: MockTopicEditorStateService;
+  let subtopicValidationService: SubtopicValidationService;
+  let topicObjectFactory: TopicObjectFactory;
+  let topic = null;
+  let DefaultSubtopicPageSchema = {
+    type: 'html',
+    ui_config: {
+      rows: 100
+    }
+  };
+
+  beforeEach(waitForAsync(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        CreateNewSubtopicModalComponent
+      ],
+      providers: [
+        {
+          provide: WindowRef,
+          useClass: MockWindowRef
+        },
+        {
+          provide: NgbActiveModal,
+          useClass: MockActiveModal
+        },
+        {
+          provide: TopicEditorStateService,
+          useClass: MockTopicEditorStateService
+        },
+        TopicUpdateService,
+        SubtopicValidationService
+      ],
+      schemas: [NO_ERRORS_SCHEMA]
+    }).compileComponents();
+  }));
+  beforeEach(() => {
+    fixture = TestBed.createComponent(CreateNewSubtopicModalComponent);
+    component = fixture.componentInstance;
+
+    ngbActiveModal = TestBed.inject(NgbActiveModal);
+    TestBed.inject(WindowRef);
+    topicUpdateService = TestBed.inject(TopicUpdateService);
+    topicEditorStateService =
+      (TestBed.inject(TopicEditorStateService) as unknown) as
+      jasmine.SpyObj<MockTopicEditorStateService>;
+    subtopicValidationService = TestBed.inject(SubtopicValidationService);
+    topicObjectFactory = TestBed.inject(TopicObjectFactory);
+
+    topic = topicObjectFactory.createInterstitialTopic();
+    let subtopic1 = Subtopic.createFromTitle(1, 'Subtopic1');
+    topic.getSubtopics = function() {
+      return subtopic1;
+    };
+    topic.getId = function() {
+      return '1';
+    };
+    topic.getNextSubtopicId = function() {
+      return 1;
+    };
+    spyOn(topicEditorStateService, 'getTopic').and.returnValue(topic);
+
+    fixture.detectChanges();
+  });
+
+  it('should assign default values to modal when initialized', () => {
+    spyOn(topicUpdateService, 'addSubtopic').and.stub();
+
+    component.ngOnInit();
+    expect(component.SUBTOPIC_PAGE_SCHEMA).toEqual(DefaultSubtopicPageSchema);
+    expect(component.subtopicId).toBe(1);
+    expect(component.hostname).toBe('local');
+    expect(component.classroomUrlFragment).toBe('non');
+    expect(component.schemaEditorIsShown).toBe(false);
+    expect(component.subtopicUrlFragmentExists).toBe(false);
+    expect(component.errorMsg).toBe(null);
+    expect(component.MAX_CHARS_IN_SUBTOPIC_TITLE)
+      .toBe(constants.MAX_CHARS_IN_SUBTOPIC_TITLE);
+    expect(component.MAX_CHARS_IN_SUBTOPIC_URL_FRAGMENT)
+      .toBe(constants.MAX_CHARS_IN_SUBTOPIC_URL_FRAGMENT);
+    expect(topicUpdateService.addSubtopic).toHaveBeenCalledWith(topic, '');
+
+    component.localValueChange('working fine');
+    expect(component.htmlData).toBe('working fine');
+  });
+
+  it('should show Schema editor when showSchemaEditor called', () => {
```

Comment:
Write the test descriptions in a way such that they reflect the user's perspective rather than including implementation details.

better description could have been `it should show schema editor when user clicks on \'Give a description or explanation of the subtopic.\' button`

Reply:
You may refer to the FE Testing Wiki, especially the naming tests part [here](https://github.com/oppia/oppia/wiki/Frontend-tests#name-tests-clearly)

Reply:
Done